plugins/modules/proxmox_cluster.py

```diff
@@ -0,0 +1,302 @@
+      type: str
+      sample: "pve2"
+    totem:
+      description: Totem protocol configuration.
+      type: dict
+      contains:
+        cluster_name:
+          description: Cluster name from totem.
+          type: str
+          sample: "devcluster"
+        config_version:
+          description: Config version.
+          type: str
+          sample: "1"
+        interface:
+          description: Interface configuration.
+          type: dict
+        ip_version:
+          description: IP version.
+          type: str
+          sample: "ipv4-6"
+        link_mode:
+          description: Link mode.
+          type: str
+          sample: "passive"
+        secauth:
+          description: Whether secure authentication is on.
+          type: str
+          sample: "on"
+        version:
+          description: Totem protocol version.
+          type: str
+          sample: "2"
+"""
+
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import missing_required_lib
+
+try:
+    from proxmoxer import ProxmoxAPI
+except ImportError:
+    PROXMOXER_LIBRARY = False
+    PROXMOXER_LIBRARY_IMPORT_ERROR = traceback.format_exc()
+else:
+    PROXMOXER_LIBRARY = True
+    PROXMOXER_LIBRARY_IMPORT_ERROR = None
+
+
+def create_cluster(proxmox, module):
+    cluster_name = module.params.get("cluster_name") or module.params.get("api_host")
+    payload = {"clustername": cluster_name}
+
+    if module.params.get("link0") is not None:
+        payload["link0"] = module.params.get("link0")
+    if module.params.get("link1") is not None:
+        payload["link1"] = module.params.get("link1")
+
+    if module.check_mode:
+        module.exit_json(changed=True, msg="Cluster '{}' would be created (check mode).".format(cluster_name), cluster=cluster_name)
+
+    try:
+        proxmox.cluster.config.post(**payload)
+    except Exception as e:
+        module.fail_json(msg="Error while creating cluster: {}".format(str(e)))
+
+    module.exit_json(changed=True, msg="Cluster '{}' created.".format(cluster_name), cluster=cluster_name)
+
+
+def get_cluster_info(proxmox, module):
+    try:
+        data = proxmox.cluster.config.join.get()
+    except Exception as e:
+        module.fail_json(msg="Error while obtaining cluster join information: {}".format(str(e)))
+
+    module.exit_json(changed=False, ansible_facts={"cluster_join_info": data})
+
+
+def join_cluster(proxmox, module):
+    master_ip = module.params.get("master_ip")
+    fingerprint = module.params.get("fingerprint")
+    api_password = module.params.get("api_password")
+    cluster_name = module.params.get("cluster_name")
+
+    if master_ip is None or fingerprint is None:
+        module.fail_json(msg="Action 'join_cluster' requires 'master_ip' and 'fingerprint' arguments.")
+
+    if module.check_mode:
+        module.exit_json(changed=True, msg="Node would join the cluster (check mode).", cluster=cluster_name)
+
+    try:
```

Comment:
Same here, will this error out when the node is already a member of _a_ cluster (AFAIK the docs mention a node should be re-installed, or at least thoroughly cleaned before joining a different cluster)?